makoctl

```diff
@@ -53,6 +55,30 @@ case "$1" in
 
 	call InvokeAction "us" "$id" "$action"
 	;;
+"menu")
+	shift 1
+	if ! type -p jq > /dev/null; then
+		echo >&2 "$0: jq is required to use 'menu'"
+		exit 1
+	fi
+	if array="$(call ListNotifications | jq -re '.data[0][0].actions.data | if length > 0 then . else false end')"; then
+		sel="$(jq -rn "$array|values[]" | "$@")"
+		sel="$(jq -rn --arg sel "$sel" "$array|to_entries[]|select(.value == \$sel).key")"
+		if [ -n "$sel" ]; then
+			# The key may contain C escape sequences, use printf %b to evaluate them.
```

Comment:
Why would the key contain escape sequences that need to be evaluated?

Reply:
It won't, my mistake. We will still need to add a trailing character to prevent newlines from being mangled.

Reply:
Honestly it's probably not worth it. Action identifiers shouldn't contain newlines.

Reply:
Fair.

Reply:
Yeah, the odds of encountering it is ridiculously small. I am inclined to guard against a possible future issue, but I'll concede.